added api calls with error handling

AliSzu opened the pull request.

Commit:
added api calls with error handling

rafalhadzicki commented on src/api/jokesApi.ts:
```diff
@@ -0,0 +1,24 @@
+export const jokesApi = {
+  fetchRandomJokes: async () => {
+    try {
+      const response = await jokesClient.get("/random");
```
please add routes as constants in separate file

rafalhadzicki commented on src/api/jokesApi.ts:
```diff
@@ -0,0 +1,24 @@
+  },
+  fetchCategories: async () => {
+    try {
+      const categories = await jokesClient.get("/categories");
```
it aplies to both endpoints: You can type it better for ex. `fetchRandomJokes: async (): Promise<AxiosResponse<JokeResponseType>>=> {
    try {
      const response = await jokesClient.get<JokeResponseType>("/random");`

rafalhadzicki commented on src/api/jokesApi.ts:
```diff
@@ -0,0 +1,24 @@
+      const categories = await jokesClient.get("/categories");
+      return categories.data;
+    } catch (error) {
+      if (error instanceof Error) {
```
you can check if error is of type AxiosResponse and return whole object with message, statuses etc. so you can handle errors better

rafalhadzicki commented on src/components/CategorySelect/CategorySelect.module.scss:
```diff
@@ -0,0 +1,82 @@
+  display: inline-block;
+}
+
+.dropdown-content {
```
you can try nesting dropdown and dropdown content https://sass-lang.com/documentation/style-rules

rafalhadzicki commented on src/components/CategorySelect/CategorySelect.tsx:
```diff
@@ -0,0 +1,67 @@
+}
+
+const CategorySelect = ({ onClick }: ICategorySelect) => {
+  const [categories, setCategories] = useState([]);
```
create enum with categories and change type to Category[]

rafalhadzicki commented on src/components/CategorySelect/CategorySelect.tsx:
```diff
@@ -0,0 +1,67 @@
+
+const CategorySelect = ({ onClick }: ICategorySelect) => {
+  const [categories, setCategories] = useState([]);
+  const [category, setCategory] = useState("Categories");
```
use created enum here

rafalhadzicki commented on src/components/CategorySelect/CategorySelect.tsx:
```diff
@@ -0,0 +1,67 @@
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [isError, setIsError] = useState(false)
+
+  const selectClasses = category === "Categories" ? "inactive" : "active"; //move this up and try to use enum instead of string or use css modules
```
use category enum here and also create enums for `inactive` and `active`

rafalhadzicki commented on src/components/CategorySelect/CategorySelect.tsx:
```diff
@@ -0,0 +1,67 @@
+  useEffect(() => {
+    const getCategories = async () => {
+      const categoriesData = await jokesApi.fetchCategories();
+      if (typeof categoriesData === "string") {
```
A few comments ago I told you to use AxiosResponse as a return item of categories data, so in this case, check if the status is ok and if it's not then set error

rafalhadzicki commented on src/components/AlertSnackbar/AlertSnackbar.tsx:
```diff
@@ -0,0 +1,28 @@
+import { useState } from "react";
+
+interface IAlertSnackbar {
+  message: string;
```
add here opening function and boolean so you can control snackbar from outside, as we delete mui, please create custom snackbar yourself

rafalhadzicki commented on src/components/CategorySelect/CategorySelect.tsx:
```diff
@@ -0,0 +1,67 @@
+
+  return (
+    <>
+    {isError && <AlertSnackbar message={errorMessage!}/>}
```
instead of using conditional rendering pass boolean and set function to this component

rafalhadzicki commented on src/components/Image/Image.tsx:
```diff
@@ -0,0 +1,20 @@
+  customName: boolean;
+}
+
+const Image = ({ customName }: IImage) => {
```
is the custom name being used here?

rafalhadzicki commented on src/components/MainScreen/MainScreen.module.scss:
```diff
@@ -0,0 +1,28 @@
+
+}
+
+.joke-text {
```
nest this inside joke if possible

rafalhadzicki commented on src/components/MainScreen/MainScreen.tsx:
```diff
@@ -0,0 +1,49 @@
+  useEffect(() => {
+    const getJoke = async () => {
+      const chuckData = await jokesApi.fetchRandomJokes();
+      if (typeof chuckData === "string") {
```
same thing as in the category comment

rafalhadzicki commented on src/components/MainScreen/MainScreen.tsx:
```diff
@@ -0,0 +1,49 @@
+  }, []);
+
+  const onCategorySelect = (category: string) => {
+    console.log(category);
```
don't code with console logs, if it's temporary you should add comment mentioning that

rafalhadzicki commented on src/components/MainScreen/MainScreen.tsx:
```diff
@@ -0,0 +1,49 @@
+
+  return (
+    <>
+      {isError && <AlertSnackbar message={errorMessage!} />}
```
Same as in Category component

rafalhadzicki commented on src/components/MainScreen/MainScreen.tsx:
```diff
@@ -0,0 +1,49 @@
+        </div>
+        <CategorySelect onClick={onCategorySelect} />
+        <div>input</div>
+        <div>button</div>
```
should be commented why it's here ex. `//placeholder`

rafalhadzicki commented on src/types/ChuckJoke.ts:
```diff
@@ -0,0 +1,10 @@
+export interface ChuckJoke {
+    categories: [];
```
use created enum here

Szulerinio commented on src/components/CategorySelect/CategorySelect.module.scss:
```diff
@@ -0,0 +1,82 @@
+  z-index: 1;
+  margin-top: -5px;
+  border: 2px solid;
+  width: -webkit-fill-available;
```
weird to use one vendor-prefixed value only
i've never used that one, but quick google search says its similar to width:100%. why not use this one instead?

AliSzu committed:
changed snackbar form mui to custom

AliSzu committed:
added enum for category

rafalhadzicki commented on src/constans/ApiPaths.ts:
```diff
@@ -0,0 +1,2 @@
+export const RANDOM_CHUCK_JOKE_PATH = '/random'
+export const CATEGORIES_PATH ='/categories'
```
i think it would be better if you created one object instead of exporting separate constants:
` const PATHS={
    RANDOM_CHUCK_JOKE_PATH: '/random',
    CATEGORIES_PATH: '/categories'
    }`

rafalhadzicki commented on src/enums/CategoryEnum.ts:
```diff
@@ -0,0 +1,19 @@
+export enum CategoryEnum {
+    categories = 'Categories',
```
enum names should should be capital `Categories='Categories'`

AliSzu committed:
fixed snackbar opening + small fixes

small fixes includes:
better styling for category select, deleted -webkit-fill-available,
added all paths to one object

AliSzu committed:
added enum for category status and fixed enum typings
